lib/segment/src/segment/snapshot.rs

```diff
@@ -338,39 +361,47 @@ pub fn snapshot_files(
         let stripped_path = strip_prefix(&file, &segment.current_path)?;
 
         if include_if(stripped_path) {
-            tar.blocking_append_file(&file, stripped_path)?;
+            tar.blocking_append_file(&file, stripped_path)
+                .map_err(|err| failed_to_add("payload index file", &file, err))?;
         }
     }
 
     for file in segment.payload_storage.borrow().files() {
         let stripped_path = strip_prefix(&file, &segment.current_path)?;
 
         if include_if(stripped_path) {
-            tar.blocking_append_file(&file, stripped_path)?;
+            tar.blocking_append_file(&file, stripped_path)
+                .map_err(|err| failed_to_add("payload storage file", &file, err))?;
         }
     }
 
     for file in segment.id_tracker.borrow().files() {
         let stripped_path = strip_prefix(&file, &segment.current_path)?;
 
         if include_if(stripped_path) {
-            tar.blocking_append_file(&file, stripped_path)?;
+            tar.blocking_append_file(&file, stripped_path)
+                .map_err(|err| failed_to_add("id tracker file", &file, err))?;
         }
     }
 
-    tar.blocking_append_file(
-        &segment.current_path.join(SEGMENT_STATE_FILE),
-        Path::new(SEGMENT_STATE_FILE),
-    )?;
+    let segment_state_path = segment.current_path.join(SEGMENT_STATE_FILE);
+    tar.blocking_append_file(&segment_state_path, Path::new(SEGMENT_STATE_FILE))
+        .map_err(|err| failed_to_add("segment state file", &segment_state_path, err))?;
 
-    tar.blocking_append_file(
-        &segment.current_path.join(VERSION_FILE),
-        Path::new(VERSION_FILE),
-    )?;
+    let version_file_path = segment.current_path.join(VERSION_FILE);
+    tar.blocking_append_file(&version_file_path, Path::new(VERSION_FILE))
+        .map_err(|err| failed_to_add("segment version file", &version_file_path, err))?;
 
     Ok(())
 }
 
+fn failed_to_add(what: &str, path: &Path, err: impl fmt::Display) -> OperationError {
+    OperationError::service_error(format!(
+        "failed to add {what} {} into snapshot: {err}",
+        path.display(),
+    ))
+}
```

Comment:
Since this is exclusively used in the function above, how do you feel about inlining it?

Reply:
I never bother with explicit inlining, but sure.